File: lemongrass/src/main/java/com/ttoannguyen/lemongrass/dto/response/group/GroupJoinResponse.java
package com.ttoannguyen.lemongrass.dto.Response.group;

import com.ttoannguyen.lemongrass.entity.enums.GroupRole;
import com.ttoannguyen.lemongrass.entity.enums.MembershipStatus;
import lombok.*;
import lombok.experimental.FieldDefaults;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class GroupJoinResponse {
  String groupId;
  String groupName;
  GroupRole groupRole;
  MembershipStatus status;
  boolean isJoined;
  boolean isPendingApproval;
}

File: lemongrass-client/src/types/group/GroupJoinResponse.ts
import type { GROUPROLE } from "../enums/grouprole.enum";
import type { JOINSTATUS } from "../enums/joinstatus.enum";

export interface GroupJoinResponse {
  groupId: string;
  groupName: string;
  groupRole: GROUPROLE;
  status: JOINSTATUS;
  joined: boolean;
  pendingApproval: boolean;
}
